app/routes/me.tsx

```diff
@@ -66,57 +74,138 @@ function YouScreen() {
+                  >
+                    connected
+                  </a>
+                ) : (
+                  <a className="text-black dark:text-white" href={authorizeURL}>
+                    Connect to Discord
+                  </a>
+                )}
+              </p>
+            </div>
+
+            <Input
+              id="discord-id"
+              name="discord"
+              value={userInfo.discord?.username ?? user.discordId ?? ''}
+              placeholder="n/a"
+              readOnly
+              className="mb-12 lg:mb-20"
+              aria-describedby="discord-message"
+            />
+
+            <Button type="submit">Save changes</Button>
           </div>
-          <button type="submit">Submit</button>
-        </Form>
-      </details>
-      <div>
-        {user.discordId ? (
-          <div>
-            Connected to discord account:{' '}
-            <a href={`https://discord.com/users/${user.discordId}`}>
-              {userInfo.discord?.username ?? user.discordId}
-            </a>
-          </div>
-        ) : (
-          <>
-            <div>You wanna connect your account to discord?</div>
-            <a href={authorizeURL}>Connect my KCD account to Discord</a>
-          </>
-        )}
-      </div>
-      <div>
-        <details>
-          <summary>Login on another device:</summary>
-          <div>
-            Scan this QR code on the other device:
-            <img src={data.qrLoginCode} alt="Login QR Code" />
+
+          <div className="col-span-full lg:col-span-4 lg:col-start-8">
+            <Label className="mb-4" htmlFor="chosen-team">
+              Chosen team
+            </Label>
+
+            <input
+              className="sr-only"
+              type="radio"
+              name="team"
+              value={user.team}
+              checked
+              readOnly
+            />
+
+            <div className="relative col-span-full mb-3 bg-gray-100 dark:bg-gray-800 rounded-lg focus-within:outline-none ring-team-current ring-offset-team-current ring-offset-team-current ring-offset-4 ring-2 focus-within:ring-2 lg:col-span-4 lg:mb-0">
+              <span className="absolute left-9 top-9 text-team-current">
+                <CheckIcon />
+              </span>
+
+              <div className="block pb-12 pt-20 px-12 text-center">
+                <img
+                  className="block mb-16"
+                  src={TEAM_MAP[user.team].image.src}
+                  alt={TEAM_MAP[user.team].image.alt}
+                />
+                <H6>{TEAM_MAP[user.team].label}</H6>
+              </div>
+            </div>
           </div>
-        </details>
-      </div>
+        </Grid>
+      </Form>
+
+      <Grid>
+        <div className="col-span-full mb-12 lg:col-span-5 lg:col-start-8 lg:mb-0">
+          <H2 className="mb-2">Need to login somewhere else?</H2>
+          <H2 variant="secondary" as="p">
+            Scan this QR code on the other device.
```

Comment:
Still waiting, but added something creative in the mean while. I think it doesn't even look that bad.

Reply:
I like the way it looks. Technically it's challenging though because I don't want the QR code to show up without the user clicking it for two reasons:

1. I want the QR code to expire after a short period
2. I don't want someone else to be able to scan it over the person's shoulder

So it should be hidden until the user clicks something and when they do we'll make a request to generate it.

Reply:
Makes sense. How about adding an overlay mentioning "click to reveal", exactly where the image is right now. And then when clicked, we fade that overlay out, and the QR in?

Reply:
Oh yeah, good idea. And maybe we could put a little 60 second countdown timer in the middle of the QR code. Once the time is up it fades back out and you have to generate a new one. We'll also want a button to hide it earlier. This is all the "fun" and "extra" stuff so if you have other things you want to get done sooner go ahead and just keep this bit out for now.

Reply:
Actually, I'll just merge this as-is. We can come back to this later.